src/spectral/utils/spectral-validator.js

```diff
@@ -0,0 +1,44 @@
+const MessageCarrier = require('../../plugins/utils/messageCarrier');
+
+const parseResults = function(results) {
+  const messages = new MessageCarrier();
+
+  if (results) {
+    for (const validationResult of results) {
+      if (validationResult) {
+        const code = validationResult['code'];
+        const severity = validationResult['severity'];
+        const message = validationResult['message'];
+        const path = validationResult['path'];
+
+        if (code && severity != null && message && path) {
+          if (code == 'parser') {
```

Comment:
Typically, you always want to use strict operators. This applies to the following checks as well

```suggestion
          if (code === 'parser') {
```